sdk/java/core/src/main/kotlin/com/keepersecurity/secretsManager/core/SecretsManager.kt

```diff
@@ -301,7 +301,31 @@ data class KeeperFile(
     val data: KeeperFileData,
     val url: String,
     val thumbnailUrl: String?
-)
+) {
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as KeeperFile
+
+        if (!fileKey.contentEquals(other.fileKey)) return false
+        if (fileUid != other.fileUid) return false
+        if (data != other.data) return false
+        if (url != other.url) return false
+        if (thumbnailUrl != other.thumbnailUrl) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = fileKey.contentHashCode()
+        result = 31 * result + fileUid.hashCode()
+        result = 31 * result + data.hashCode()
+        result = 31 * result + url.hashCode()
+        result = 31 * result + (thumbnailUrl?.hashCode() ?: 0)
+        return result
+    }
+}
```

Comment:
We have a bytearray in this data class which means we need to implement our own equals and hashcode

Reply:
No, we don't do equality checks on this class

Reply:
That hashcode is more important here - if this ever goes in a hashed data structure without it it won't work